### PR TITLE
adding new alternative quiz component

### PR DESCRIPTION
**Pull request by mendelB**
### What does this PR do?
This PR adds a new alternative Quiz component suite featuring two primary changes for now from the legacy Quiz:

- quiz choice selections stored in local state, enabling a way less cluttered layout with no reducers or actions necessary.
- visual changes based on the design specs (attached to the pivotal card)
- also, instead of displaying an error when the quiz is submitted incomplete, we just disable the submit button until all questions are answered 💡 

Beyond that, it's more or less following a similar format with the main `QuizAlt` component that renders `QuizQuestion`s which each render `QuizChoice`s, along with a top-level `QuizConclusion` which is used as the CTA/Quiz submission button. 

The main thing missing at this point is the award system based on the selections, as well as the corresponding Action. 

📸 
[**Small**](https://user-images.githubusercontent.com/12417657/37618065-44483afc-2b8b-11e8-9748-116bb37c9380.png)

[**Medium**](https://user-images.githubusercontent.com/12417657/37616432-4430b7c4-2b86-11e8-81cf-958e6b1c17c5.png)

[**Large** incomplete quiz](https://user-images.githubusercontent.com/12417657/37616361-fb045aa6-2b85-11e8-8d2a-0ec0a013117b.png)
[**Large** complete quiz](https://user-images.githubusercontent.com/12417657/37616378-18588942-2b86-11e8-9c6f-a0bb9c889c1b.png)

### What are the relevant tickets/cards?
[Pivotal ref #155783121](https://www.pivotaltracker.com/story/show/155783121)

### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [x] Added screenshot to PR description of related front-end updates on **small** screens.
- [x] Added screenshot to PR description of related front-end updates on **medium** screens.
- [x] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by weerd** [comment]
> quiz choice selections stored in local state, enabling a way less cluttered layout with no reducers or actions necessary.

🎉

File: resources/assets/components/Campaign/Campaign.js [hunk 1]
```diff
@@ -36,7 +37,7 @@ const Campaign = props => (
     ) : null }
 
     { props.shouldShowLandingPage ?
-      <LandingPageContainer {...props} />
+      <QuizAltContainer />
```
Comment (weerd) [inline resources/assets/components/Campaign/Campaign.js]: Why are we replacing the `LandingPage` with the `Quiz`?

Comment (weerd) [reply resources/assets/components/Campaign/Campaign.js]: temp?

**Comment by weerd** [comment]
Could we rename the _old_ Quiz components as `LegacyQuiz` instead of `QuizAlt` for the new stuff? Makes it easier to just delete the old stuff when ready.

File: resources/assets/components/Campaign/Campaign.js (hunk 1 above)
Comment (mendelB) [reply resources/assets/components/Campaign/Campaign.js]: Just temporary for testing on the deploy page imma delete before merging.

**Comment by mendelB** [comment]
> Could we rename the old Quiz components as LegacyQuiz instead of QuizAlt for the new stuff? Makes it easier to just delete the old stuff when ready.

Happily! `QuizAlt` was driving me crazy tbh 😬

**Comment by weerd** [comment]
Haha! @mendelB if it's too much for this PR. Maybe just in an quick follow up PR after merging this do the rename? Up to you!

File: resources/assets/components/QuizAlt/QuizAlt.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  constructor() {
+    super();
+
+    this.state = {
+      choices: {},
+    };
+
+    this.selectChoice = this.selectChoice.bind(this);
+    this.completedQuiz = this.completedQuiz.bind(this);
+    this.completeQuiz = this.completeQuiz.bind(this);
```
Comment (weerd) [inline resources/assets/components/QuizAlt/QuizAlt.js]: Quick note for a future refactor! I updated our `bable-config` so now we have access to class properties! Which means you probs don't need a constructor anymore. You can declare the `state` as a property of the class as well as the methods, which autobind! I'll send you a screenshot via Slack for something I'm working on that is using all this.

File: resources/assets/components/QuizAlt/QuizAlt.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  selectChoice(questionId, choiceId) {
+    const choices = Object.assign({}, this.state.choices);
+    choices[questionId] = choiceId;
+    this.setState({ choices });
+  }
+
+  completedQuiz() {
+    return every(this.props.questions, question => (
```
Comment (weerd) [inline resources/assets/components/QuizAlt/QuizAlt.js]: `every()` very nice! 👌

File: resources/assets/components/QuizAlt/QuizAlt.js (hunk 2 above)
Comment (mendelB) [reply resources/assets/components/QuizAlt/QuizAlt.js]: ahhhhhh that is sick 🔥

File: resources/assets/components/QuizAlt/QuizQuestion.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      key={choice.id}
+      questionId={id}
+      selectChoice={selectChoice}
+    />
+  ));
+
+  return (
+    <div className="question">
+      <SectionHeader
+        preTitle={`Question ${convertNumberToWord(Number(id) + 1)}`}
```
Comment (weerd) [inline resources/assets/components/QuizAlt/QuizQuestion.js]: random note, we're gonna go with `superTitle` so it follows well with `subTitle` around the title; "super script" and "sub script" :)

**Comment by mendelB** [comment]
Ok got rid of the weird stuff!

**Comment by mendelB** [comment]
refs #780

**Pull request merged**